[DOC] Adjust some new features wording/examples.

zverok opened the pull request:
(Part of working on this year's changelog.)

Some adjustments of the wording of the new features docs.

The changes are relatively small, so I packed them into one PR. If this is unacceptable, I can split in several (every change is its own commit), but I hope that it can be reviewed easily.

The changes in this PR:
* [Range#overlap?](https://docs.ruby-lang.org/en/master/Range.html#method-i-overlap-3F): changed wording of the last paragraph from less packed and extracted inline code as examples.
* [Thread::Queue#freeze](https://docs.ruby-lang.org/en/master/Thread/Queue.html#method-i-freeze): adjusted method docs with explanations why the queue can't be frozen.
* [TracePoint](https://docs.ruby-lang.org/en/master/TracePoint.html): added missing new `:rescue` event.
* [Module#set_temporary_name](https://docs.ruby-lang.org/en/master/Module.html#method-i-set_temporary_name): slightly expanded examples and explanations with more visible consequences and possible usecases.
* [Process::Status#&](https://docs.ruby-lang.org/en/master/Process/Status.html#method-i-26) and [#>>](https://docs.ruby-lang.org/en/master/Process/Status.html#method-i-3E-3E): slightly expanded deprecation notice, removed `>>` usage from class header example (as it is deprecated, we shoudn't encourage it with the initial example).

Added Dec 12:
* [MatchData#named_captures](https://docs.ruby-lang.org/en/master/MatchData.html#method-i-named_captures): Fix rendering glitch due to wrong indentation.
* [Refinement#target](https://docs.ruby-lang.org/en/master/Refinement.html#method-i-target): Adjust docs a bit, add deprecation notice to `#refined_class`
* [Dir#fchdir](https://docs.ruby-lang.org/en/master/Dir.html#method-c-fchdir): Fix examples (they were confusing: the first one, after the description of the no-block form, demonstrated only a block form; the second one didn't demonstrate `#fchdir` at all)

jeremyevans commented on thread_sync.c:
```diff
@@ -1163,8 +1163,13 @@ NORETURN(static VALUE rb_queue_freeze(VALUE self));
+ * The queue can't be frozen, so this method raises an exception:
+ *   Thread::Queue.new.freeze # Raises TypeError (cannot freeze #<Thread::Queue:0x...>)
+ *
+ * Unlike other collections, freezing a queue (to stop accepting #pop and #push)
```
I don't think the documentation needs to justify the implementation, only explain it, so I would remove this second paragraph.

jeremyevans commented on trace_point.rb:
```diff
@@ -396,7 +397,7 @@ def return_value
   end
 
-  # Value from exception raised on the +:raise+ event
+  # Value from exception raised on the +:raise+ event, or rescued on the +rescue+ event.
```
For consistency, either remove the colon before `raise`, or add it before `rescue`.

jeremyevans commented on variable.c:
```diff
@@ -162,21 +162,21 @@ is_constant_path(VALUE name)
  *
- *  If the name given is not a string or is a zero length string, this method
- *  raises an ArgumentError.
+ *  The method can be useful to name classes and modules auto-generated from
```
I would remove this paragraph. "The DSL" is ambiguous, "RSpec" is not in ruby core or stdlib, "Web-framework routes" could only use this if they were classes or modules.

jeremyevans commented on variable.c:
```diff
@@ -189,15 +189,27 @@ is_constant_path(VALUE name)
+ *
+ *    c.new # => #<MyClass(with description):0x0....>
+ *    c.instance_method(:foo) # => #<UnboundMethod: MyClass(with description)#foo() ...>
+ *    c.new.method(:foo) #<Method: MyClass(with description)#foo() ...>
```
Missing `# =>` in this line

zverok replied:
@mame My point was to demonstrate that _all_ related objects are affected (as an emphasis on "why would this be useful").
Though we probably can just leave one of the two lines (`instance_method` vs `method`), did you mean that?

nobu commented:
FYI: `Docs:` tags do nothing for us.  `[DOC]` is to suppress CIs.

mame commented on range.c:
```diff
@@ -2391,18 +2391,17 @@ empty_region_p(VALUE beg, VALUE end, int excl)
- *  where +minimum+ is a value that no value is smaller than.
- *  Such values include <tt>-Float::INFINITY</tt>, <tt>[]</tt>, <tt>""</tt>, and
- *  classes without subclasses.
+ *  Note that the method wouldn't make any assumptions about the semi-open
```
Rather than introducing a new term "semi-open range," I would prefer "exclusive range," which is used elsewhere.

zverok replied:
I used it here as a synonym for "beginless or endless" (not for "exclusive range"), yet the statement is related only to beginless one, so I'll use more precise wording.

zverok commented:
@jeremyevans I tried to address all of your concerns. I also added 3 more small commits (listed under `Dec 12` in the PR description), I hope that's OK for "fixes a bit of everything in docs" PR :shrug:

jeremyevans commented on eval.c:
```diff
@@ -1346,9 +1346,19 @@ rb_using_module(const rb_cref_t *cref, VALUE module)
+ *
+ *     module M
+ *       refine String do
+ *         def indent(level)
```
Best remove the method definition here, it doesn't affect the result and is unrelated to the method being documented.

jeremyevans commented on variable.c:
```diff
@@ -189,15 +189,25 @@ is_constant_path(VALUE name)
  *
- *    n = Module.new
- *    n.set_temporary_name("fake_name")
+ *    c = Class.new do
```
I'm with @mame here, the method already has significant examples, I don't think we need to add to them.

zverok replied:
OK. My point here was to demonstrate one of the cases where redefining `Module#name` and/or `Module#inspect` wouldn't fix the related object's `#inspect` to be meaningful (and adds to a justification "why somebody needs a new method to set a pretend-name"). But probably, it doesn't look important enough anyway.

jeremyevans reviewed (changes requested):
Looks almost ready.  Just a couple changes requested.

zverok commented:
@jeremyevans I think I fixed all the comments. So I rebased it from `master`, and squashed all the fixups so it should be ready to merge.

Thanks for the reviews!

jeremyevans reviewed (approved):
Please squash when merging.

zverok commented:
@jeremyevans Do you mean we need to have just one commit with all the changes (because I tried to structure it in a "commit-per-class" fashion)?

jeremyevans commented:
Yes.
